lib/web_ui/lib/src/ui/initialization.dart

```diff
@@ -39,24 +39,24 @@ Future<void> _initializePlatform({
   if (engine.experimentalUseSkia) {
     await engine.skiaFontCollection.ensureFontsLoaded();
   } else {
-    await _fontCollection.ensureFontsLoaded();
+    await _fontCollection!.ensureFontsLoaded();
   }
 
   _webOnlyIsInitialized = true;
 }
 
 // TODO(yjbanov): can we make this late non-null? See https://github.com/dart-lang/sdk/issues/42214
-engine.AssetManager/*?*/ _assetManager;
-engine.FontCollection _fontCollection;
+engine.AssetManager? _assetManager;
+engine.FontCollection? _fontCollection;
```

Comment:
Should this be a `late`?

Reply:
Haven't been lucky with late yet, but maybe I'm not trying hard enough. The issue is most of our initialization patterns rely on null checks, e.g. `if (foo == null) initializeFoo();`. However, `late` non-null variables cannot be null checked.

It's OK for this PR to be incomplete. The goal is to move to the new syntax and to have a more or less correct public API surface. Internal parts can be revisited later.

Reply:
Filed https://github.com/flutter/flutter/issues/59371 to revisit this.

Reply:
Looks to me like it could be late final engine.FontCollection _fontCollection = engine.FontCollection();